### PR TITLE
`fix`: added Setup.sh to be MacOS Friendly

### PR DESCRIPTION
**Pull request by Ntropy86**
For Macos users made the `setup.sh` file more friendly to use.
Issue Fixed: In Apple Silicon Macs, The CUDA version does not work, it requires MPS version.
Extra Fixes: 
- Added Auto detection OS and installs the correct files
- Followed Literal Definitions standards
- Added Comments

### CODE REVIEW

**Comment by nckcard** [comment]
Hey, this looks great, thanks. One concern: Pytorch CTC loss is not supported on MPS (as far as I know), which would prevent someone using this repo from running the baseline model training on MPS. Have you tried doing so on MPS / is there anyway around this limitation other than using the much-slower CPU instead?

**Comment by Ntropy86** [comment]
@nckcard You are correct. It is not supported (I looked around and tried to see if there are any other solutions for CTC, did not find anything promising. So what I can do is add a WARN for macos users and make the run Hybrid approach using MPS normally but Running CTC on CPU.

Upon reading more, I believe we can use Attention Loss however that would be a massive architectural change and not a direct implementation of the paper.

**Comment by nckcard** [comment]
If you can make it run hybrid, that seems fine to me. As is, I think the training script will crash if we try to train with CTC loss via MPS. Attention loss also seems very interesting, if you want to PR that separately I would be supportive! Thanks.
